ci-operator/step-registry/hypershift/kubevirt/create/hypershift-kubevirt-create-chain.yaml

```diff
@@ -0,0 +1,136 @@
+chain:
+  as: hypershift-kubevirt-create
+  steps:
+  - as: create-hostedcluster
+    cli: latest
+    env:
+    - name: HYPERSHIFT_NODE_COUNT
+      default: "3"
+      documentation: "The number nodes to automatically create and join to the cluster."
+    - name: HYPERSHIFT_BASE_DOMAIN
+      default: "ci.azure.devcluster.openshift.com"
```

Comment:
For this dns trick to work, the basedomain for the hypershift cluster needs to be a subdomain of the infra's base domain.

So, if the infra cluster is named "my-infra" and has the base domain `ci.azure.devcluster.openshift.com`, then the hypershift's basedomain is actually `apps.my-infra.ci.azure.devcluster.openshift.com`

You can automate the detection of this by doing something like this in the infra cluster.

`oc get ingresscontroller -n openshift-ingress-operator default -o yaml | grep domain | awk '{print $2 }'` 

That's going to return the `apps.<infra cluster name>.<base domain>` which is the basedomain that should be used for the hypershift guest cluster.

Reply:
I removed this default env var as it getting dynamically discovered from `ingresscontroller` CR, as suggested.